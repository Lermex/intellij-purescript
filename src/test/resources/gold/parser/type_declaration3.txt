Purescript File
  PSASTWrapperElement(Program)
    PSASTWrapperElement(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSASTWrapperElement(ModuleName)
        PSASTWrapperElement(Qualified)
          PsiElement(proper name)('Main')
      PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(TypeSynonymDeclaration)
        PsiElement(type)('type')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeConstructor)
          PsiElement(proper name)('Model')
          PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)('\n  ')
        PSASTWrapperElement(Type)
          PSASTWrapperElement(TypeAtom)
            PSASTWrapperElement(ObjectType)
              PsiElement({)('{')
              PsiElement(whitespace)('\n    ')
              PSASTWrapperElement(Row)
                PSASTWrapperElement(LocalIdentifier)
                  PSASTWrapperElement(identifier)
                    PsiElement(identifier)('users')
                  PsiElement(whitespace)(' ')
                PsiElement(::)('::')
                PsiElement(whitespace)(' ')
                PSASTWrapperElement(Type)
                  PSASTWrapperElement(TypeAtom)
                    PSASTWrapperElement(TypeConstructor)
                      PSASTWrapperElement(Qualified)
                        PsiElement(proper name)('Array')
                        PsiElement(whitespace)(' ')
                  PSASTWrapperElement(TypeAtom)
                    PSASTWrapperElement(TypeConstructor)
                      PSASTWrapperElement(Qualified)
                        PsiElement(proper name)('User')
                        PsiElement(whitespace)('\n  ')
              PsiElement(})('}')