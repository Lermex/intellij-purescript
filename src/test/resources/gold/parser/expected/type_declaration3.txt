Purescript File
  PSModuleDeclarationImpl(MODULE_DECLARATION)
    PsiElement(PSTokenType.module)('module')
    PSModuleNameImpl(MODULE_NAME)
      PsiElement(PSTokenType.PROPER_NAME)('Main')
    PsiElement(PSTokenType.where)('where')
  PSDeclarationsImpl(DECLARATIONS)
    PSDeclarationImpl(DECLARATION)
      PSTypeDeclarationImpl(TYPE_DECLARATION)
        PsiElement(PSTokenType.type)('type')
        PSDeclarationRefTypeListImpl(DECLARATION_REF_TYPE_LIST)
          PSDeclarationRefTypeImpl(DECLARATION_REF_TYPE)
            PsiElement(PSTokenType.PROPER_NAME)('Model')
        PsiElement(PSTokenType.=)('=')
        PSTypeDeclarationItemImpl(TYPE_DECLARATION_ITEM)
          PSDataDeclarationBodyImpl(DATA_DECLARATION_BODY)
            PsiElement(PSTokenType.{)('{')
            PSDataDeclarationTypeDefinitionListImpl(DATA_DECLARATION_TYPE_DEFINITION_LIST)
              PSDataDeclarationTypeDefinitionImpl(DATA_DECLARATION_TYPE_DEFINITION)
                PsiElement(PSTokenType.ID)('users')
                PsiElement(PSTokenType.::)('::')
                PSParenthesisedTypeThenIdListImpl(PARENTHESISED_TYPE_THEN_ID_LIST)
                  PSParenthesisedDeclarationRefTypeListImpl(PARENTHESISED_DECLARATION_REF_TYPE_LIST)
                    PSDeclarationRefTypeImpl(DECLARATION_REF_TYPE)
                      PsiElement(PSTokenType.PROPER_NAME)('Array')
                    PSParenthesisedDeclarationRefTypeListImpl(PARENTHESISED_DECLARATION_REF_TYPE_LIST)
                      PSDeclarationRefTypeImpl(DECLARATION_REF_TYPE)
                        PsiElement(PSTokenType.PROPER_NAME)('User')
            PsiElement(PSTokenType.})('}')